### PR TITLE
Fix for Bug 2142619

### PR DESCRIPTION
**Pull request by avirupch**

### CODE REVIEW

File: src/ServiceManagement/StorSimple/Commands.StorSimple/Cmdlets/DataContainer/NewAzureStorSimpleDeviceVolumeContainer.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ public override void ExecuteCmdlet()
                     return;
                 }
 
-                if(EncryptionEnabled == true && string.IsNullOrEmpty(EncryptionKey))
+                if(EncryptionEnabled == true && (string.IsNullOrEmpty(EncryptionKey) || !IsValidAsciiString(EncryptionKey)))
```
Comment (Sumanth-SK) [inline src/ServiceManagement/StorSimple/Commands.StorSimple/Cmdlets/DataContainer/NewAzureStorSimpleDeviceVolumeContainer.cs]: check for validAscii in a different if statement, as this one returns an Arg null exception

File: src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs [hunk 2]
```diff
@@ -512,5 +513,10 @@ internal bool ValidTargetDeviceForFailover(string sourceDeviceId, string targetD
             
             return true;
         }
+
+        internal bool IsValidAsciiString(string s)
+        {
+            return Regex.IsMatch(s, "[ -~]+");
```
Comment (Sumanth-SK) [inline src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs]: Please use a regex instance object (instantiated once and shared), as it is much faster than a static regex match

File: src/ServiceManagement/StorSimple/Commands.StorSimple/Cmdlets/DataContainer/NewAzureStorSimpleDeviceVolumeContainer.cs (hunk 1 above)
Comment (avirupch) [inline src/ServiceManagement/StorSimple/Commands.StorSimple/Cmdlets/DataContainer/NewAzureStorSimpleDeviceVolumeContainer.cs]: Will fix

File: src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs (hunk 2 above)
Comment (avirupch) [inline src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs]: Where do you want to initialize the regex instance? If it is initialized for each cmdlet, then will there be any performance improvement?

Comment (Sumanth-SK) [inline src/ServiceManagement/StorSimple/Commands.StorSimple/StorSimpleCmdletBase.cs]: sorry, you may skip this comment. 
Just found out that the static regex match call actually caches the regex tree, so not much perf difference between creating the instance object and using the static regex call

**Pull request merged**